modules/standard/CTypes.chpl

```diff
@@ -674,9 +695,13 @@ module CTypes {
   inline operator ==(a: _nilType, b: c_ptr) {
     return __primitive("ptr_eq", a, b);
   }
+  // because chpl_c_string and c_ptrConst(c_char) can coerce back and forth,
+  // restrict this equality operator so we don't go looking at the eltType of
+  // a chpl_c_string as that will crash the compiler - whoopsie
   @chpldoc.nodoc
   inline operator c_ptrConst.==(a: c_ptrConst, b: c_ptrConst)
-      where a.eltType == b.eltType || a.eltType == void || b.eltType == void {
+      where a.type != chpl_c_string && b.type != chpl_c_string &&
```

Comment:
This doesn't make sense to me, since if `chpl_c_string` is a different type from `c_ptrConst(c_char)`, when we instantiate this function, it can't end up with `chpl_c_string` when implicitly converting. Is this still needed?

Reply:
following advice in https://github.com/chapel-lang/chapel/pull/22622#discussion_r1283709792 made this addition to the `where` clause unnecessary